filebeat/prospector/prospector_log.go

```diff
@@ -112,17 +114,69 @@ func (l *Log) Run() {
 	}
 }
 
+// globPattern detects the use of "**" and expands it to standard glob patterns up to a max depth
+func (l *Log) globPatterns(pattern string) ([]string, error) {
```

Comment:
Perhaps we could move this to a separate package and pass the max depth to the function. Reason is like this we could also use it in other places and it does not further "bloat" the prospector code.